src/parsing/medium.py

```diff
@@ -795,27 +804,31 @@ async def estimate_message_counts(self):
         return sum(1 for _ in media[0])
 
     def __len__(self) -> int:
-        return sum(1 for medium in self._media if not medium.drop_silently)
+        return sum(not medium.drop_silently for medium in self._media)
 
     def __bool__(self) -> bool:
         return any(not medium.drop_silently for medium in self._media)
 
     @property
     def valid_count(self):
-        return sum(1 for medium in self._media if medium.valid and not medium.drop_silently)
+        return sum(medium.valid and not medium.drop_silently for medium in self._media)
```

Comment:
https://github.com/Rongronggg9/RSS-to-Telegram-Bot/blob/1365e320c25ac25330536ad6515cfcc6f042a2bc/src/parsing/medium.py#L87

`None` cannot be cast to an `int`.

Reply:
```Python
return sum(medium.valid is True and not medium.drop_silently for medium in self._media)
```